src/main/java/am/ik/yavi/core/Constraint.java

```diff
@@ -51,8 +51,10 @@ default C equalTo(@Nullable V other) {
 	 * @since 0.10.0
 	 */
 	default C oneOf(Collection<V> values) {
-		this.predicates().add(ConstraintPredicate.of(values::contains, OBJECT_ONE_OF,
-				() -> new Object[] { values.toArray() }, NullAs.INVALID));
+		this.predicates()
+				.add(ConstraintPredicate.of(values::contains, OBJECT_ONE_OF,
+						() -> new Object[] { new ArrayArgument(values.toArray()) },
```

Comment:
If you prefer a simpler solution, calling `Arrays.toString(values.toArray())` works as well. 
However, it is less flexible when you need the array later. 
In that case, you'll have to parse the toString output to recreate the array.

The advantage of this solution is that the original array can be retrieved without having the need of parsing the toString.

Reply:
Maybe calling Arrays.asList is the simplest?

Reply:
Even that is not necessary, it is sufficient to simply pass the incoming collection as argument. 
I changed the test a little bit to test oneOf both with a List and a Set.

Reply:
thanks!